src/ExerciseDispatcher.php

```diff
@@ -151,6 +151,9 @@ public function verify(ExerciseInterface $exercise, Input $input): ResultAggrega
 
         try {
             $this->results->add($runner->verify($input));
+        } catch (\Throwable $e) {
```

Comment:
an added dispatch, works during tests while the application tear down ones don't.

Reply:
Not sure I understand this one? should that dispatch in the `finally` clause always get executed?

Reply:
yeah but I guess it was more of an exception handler, e.g. we already revert successfully when nothing goes wrong, but this was explicitly for cleanup on exceptions ... granted listening to `verify.post.execute` will achieve the same thing I was thinking of potentially other things hooking in for cleanup too then an explicit exception event maybe nice... I'm not overly attached to this though so happy to switch it out. 

fwiw hooking in here is important (regardless of which event we choose) as if an exception happens in doVerify then we don't hit `cli.verify.finish` thus don't currently revert.

Reply:
hmm maybe the revert should be on this instead of `cli.verify.finish` then? I think we don't really need this event. If we find a need we can always add it then.